### PR TITLE
feat(distro-ddl): add column order index for Api Plans so users can explicitly order plans in UI

### PR DESCRIPTION
**Pull request by msavy**
@volkflo initial version with UI -- needs a little tidying up + tests before committing, but seems to work well. 

Sorry for slowness due to holidays.

### CODE REVIEW

**Review by volkflo — COMMENTED** [review]
Even if you already merged this, just a question for better understanding:

The reason for not using the new rest endpoint (`/reorderApiPlans`) from the UI is that it is easier on that page to just update the whole api version?

File: manager/api/exceptions/src/main/java/io/apiman/manager/api/rest/exceptions/util/ExceptionFactory.java [hunk 1]
```diff
@@ -220,6 +220,10 @@ public static ApiVersionNotFoundException apiVersionNotFoundException(String api
         return new ApiVersionNotFoundException(Messages.i18n.format("ApiVersionDoesNotExist", apiId, version)); //$NON-NLS-1$
     }
 
+    public static ApiPlanNotFoundException apiPlanNotFoundException(String apiId, String apiVersion, String planId, String planVersion) {
+        return new ApiPlanNotFoundException(Messages.i18n.format("ApiVersionDoesNotExist", apiId, planId, planVersion)); //$NON-NLS-1$
```
[the remark this answers is not in the document]
Comment (msavy) [reply manager/api/exceptions/src/main/java/io/apiman/manager/api/rest/exceptions/util/ExceptionFactory.java:224]: Well spotted. Fixing now.

**Comment by msavy** [comment]
> Even if you already merged this, just a question for better understanding:
> 
> The reason for not using the new rest endpoint (`/reorderApiPlans`) from the UI is that it is easier on that page to just update the whole api version?

It is simply that the UI was already using the existing endpoint in several locations and I could avoid changing any code dealing with that. But to keep the API consistent with the policy chain reordering, and for other use-cases (e.g. CLI), I think it's best to add the reorder endpoint also.

**Comment by volkflo** [comment]
Okay cool, thanks for the comments :rocket:

**Pull request merged**